src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -1451,6 +1463,14 @@ public static unsafe SocketError SetSockOpt(SafeSocketHandle handle, SocketOptio
                 }
             }
 
+            if (optionLevel == SocketOptionLevel.Tcp && (int)optionName == 15)
+            {
+                handle.TfoEnabled = optionValue != 0;
+                // Silently ignore errors - TFO is best effor and it may be disabled by configuration or not
+                // supported by OS.
```

Comment:
Do we do this with other socket options? What about the getsockopt part?

Reply:
I updated the logic to be macOS specific to provide consistent experience.